### PR TITLE
feat: add webassembly structure/resources/docs

### PR DESCRIPTION
**Pull request by xytian315**
Description: 
1. set up webassembly structure/resources etc.
2. created doc for quick start: compile wasm and start web server
3. moved the function decodeJWT to lib folder for shared code 
4. renamed go module

Note: file wasm_exect.js is created by official go (https://github.com/golang/go/wiki/WebAssembly#getting-started)

### CODE REVIEW

File: docs/webassembly.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+cd wasm
+GOOS=js GOARCH=wasm go build -o ./assets/jwt.wasm .
+```
+
+1. Run go web server
+
+```bash
+cd server
+go run main.go
+```
```
Comment (ilrudie) [inline docs/webassembly.md]: **Suggestion:** Could you add these tasks to the Makefile?

File: lib/decode_jwt.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package lib
```
Comment (ilrudie) [inline lib/decode_jwt.go]: **Question:** Should we use a more descriptive name?

**Suggestion:** `package lib` -> `package jwt` or `package decode` or `package decode-jwt`

**Review by ilrudie — COMMENTED** [review]
This is cool. Couple of suggestions so far. Didn't get to run it yet but will take a look at that tomorrow <3

**Comment by ilrudie** [comment]
- [X] **Question:** `wasm/main.go` has an error in my vscode due to the platform settings being Linux/x86 on my machine instead of js/wasm as this specific file expects. Did you encounter this and find a solution?
- [ ] **Question:** should we include these vscode files in .gitignore but include some documentation or maybe just commit the files to the project?

**Note:** I was able to solve this with [vscode multi-root workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces)

File: lib/decode_jwt.go (hunk 2 above)
Comment (xytian315) [reply lib/decode_jwt.go]: renamed

File: docs/webassembly.md (hunk 1 above)
Comment (xytian315) [reply docs/webassembly.md]: added make target

**Review by ilrudie — APPROVED** [review]
Looks good to me!

Thanks!

**Pull request merged**